### ISSUE TITLE
27.6.2 update deletes master password of pwstore

### ISSUE THREAD

**Issue by computersalat** [MoonchildProductions/Pale-Moon#1529]
after upgrade to 27.6.2 I wondered why there is no 'popup' asking for the master passphrase to open the pwstore ... so I checked setting and 'using master password' was disabled.

this is not nice.
Does someone see the same ?

**Comment by wolfbeast** [comment]
Nothing has changed in that respect, and master password works just fine here.
Also, it's not possible to disable the use of the master password without being prompted for your master password to do so (they have to be decrypted), so it has to have been a deliberate user action to disable that option -- unless the entire password store was lost (which I doubt, you'd notice if there were no passwords at all).

**Comment by computersalat** [comment]
I did not say that `master password` is NOT working.
I said that USING `master password` was `disabled` during install of update. And I am not a Alzheimer's patient that I did forget that I disabled it.
You think it is impossible that it can happen, cause of your explanation. But it did happen.

**Comment by wolfbeast** [comment]
I'm saying it's impossible for that option to be disabled without entering the master password first.
If you force the option disabled without entering the master password, all your passwords will be lost. There's no two ways about it.

To top it off:
- I use a master password myself.
- I updated normally, nothing happened to the setting for me or anyone else that I know of.
- None of that code was touched at all recently.

So, I'm saying it's not possible (without being user-initiated) either way, unless you have a good explanation despite all these points? Even if it was a browser bug that would flip the option off when upgrading, see above, without the master password being entered to decrypt the password store, the passwords would not be accessible from that point forward.

The only realistic scenarios are:
1. It was user initiated (either by yourself or someone else who knows your master password). You'd have the behavior you describe.
2. There was physical corruption of the password store, and you lost all your stored passwords (including master password)
3. Your profile, through third-party intervention, was reverted to a state where the master password was not yet set
4. (extremely unlikely) You have malware on your system that specifically targets Pale Moon for stealing passwords, that simulated point 1.
